etl/tests/integration/bigquery_destination_test.rs

```diff
@@ -485,3 +490,1003 @@ async fn table_truncate_with_batching() {
+        ),
+    ];
+    let updated_timestamptz_arr = vec![
+        DateTime::<Utc>::from_naive_utc_and_offset(updated_timestamp_arr[0], Utc),
+        DateTime::<Utc>::from_naive_utc_and_offset(updated_timestamp_arr[1], Utc),
+    ];
+    let updated_uuid_arr = vec![uuid::Uuid::new_v4(), uuid::Uuid::new_v4()];
+    let updated_json_arr = vec![
+        serde_json::json!({"updated": "json1"}),
+        serde_json::json!({"updated": "json2"}),
+    ];
+    let updated_jsonb_arr = vec![
+        serde_json::json!({"updated": "jsonb1"}),
+        serde_json::json!({"updated": "jsonb2"}),
+    ];
+    let updated_oid_arr = vec![98765u32, 54321u32];
+
+    database
+        .update_values(
+            table_name.clone(),
+            &[
+                "b_arr", "t_arr", "i2_arr", "i4_arr", "i8_arr", "f4_arr", "f8_arr", "n_arr",
+                "by_arr", "d_arr", "ti_arr", "ts_arr", "tstz_arr", "u_arr", "j_arr", "jb_arr",
+                "o_arr",
+            ],
+            &[
+                &updated_bool_arr,
+                &updated_text_arr,
+                &updated_i2_arr,
+                &updated_i4_arr,
+                &updated_i8_arr,
+                &updated_f4_arr,
+                &updated_f8_arr,
+                &updated_n_arr,
+                &updated_bytes_arr,
+                &updated_date_arr,
+                &updated_time_arr,
+                &updated_timestamp_arr,
+                &updated_timestamptz_arr,
+                &updated_uuid_arr,
+                &updated_json_arr,
+                &updated_jsonb_arr,
+                &updated_oid_arr,
+            ],
+        )
+        .await
+        .unwrap();
+
+    event_notify.notified().await;
+
+    let table_rows = bigquery_database
+        .query_table(table_name.clone())
+        .await
+        .unwrap();
+    let parsed_table_rows = parse_bigquery_table_rows::<NullableColsArray>(table_rows);
+
+    let expected_updated_row = NullableColsArray::with_non_null_values(
+        1,
+        updated_bool_arr,
+        updated_text_arr,
+        updated_i2_arr,
+        updated_i4_arr,
+        updated_i8_arr,
+        updated_f4_arr,
+        updated_f8_arr,
+        updated_n_arr,
+        updated_bytes_arr,
+        updated_date_arr,
+        updated_time_arr,
+        updated_timestamp_arr,
+        updated_timestamptz_arr,
+        updated_uuid_arr,
+        updated_json_arr,
+        updated_jsonb_arr,
+        updated_oid_arr,
+    );
+
+    assert_eq!(parsed_table_rows, vec![expected_updated_row]);
+
+    // delete
+    let event_notify = destination
+        .wait_for_events_count(vec![(EventType::Delete, 1)])
+        .await;
+
+    database
+        .delete_values(table_name.clone(), &["id"], &["'1'"], "")
+        .await
+        .unwrap();
+
+    event_notify.notified().await;
+
+    let table_rows = bigquery_database.query_table(table_name.clone()).await;
+    assert!(table_rows.is_none());
+
+    pipeline.shutdown_and_wait().await.unwrap();
+}
```

Comment:
Added exhaustive tests for nullable/non-nullable scalar/array types being sent to BigQuery. These still don't exhaustively cover the range of values we'd like. Especially missing are values at the boundaries, but it's a start.